services/ui-src/src/utils/forms/dynamicItemFields.ts

```diff
@@ -0,0 +1,131 @@
+import { AnyObject, FormJson } from "types";
+
+// first attempt to combine ilos and plans functions
+
+export const generateDrawerItemFields = (
+  form: FormJson,
+  items: AnyObject[],
+  entityType: string
+) => {
+  const fields = form.fields[0];
+  return {
+    ...form,
+    fields: [
+      {
+        ...form.fields[0],
+        props: {
+          ...form.fields[0].props,
+          choices: [
+            ...updatedItemChoiceList(
+              fields.props?.choices,
+              availableItems(items, entityType),
+              entityType
+            ),
+          ],
+        },
+      },
+    ],
+  };
+};
+
+export const generateAddEntityDrawerItemFields = (
+  form: FormJson,
+  items: AnyObject[],
+  entityType: string
+) => {
+  form.fields[3] = {
+    ...form.fields[3],
+    props: {
+      ...form.fields[3].props,
+      choices: [...availableItems(items, entityType)],
+    },
+  };
+  return form;
+};
+
+export const parentFieldName = (entityType: string) => {
+  if (entityType === "plan") {
+    return "analysis_applicable";
+  } else {
+    return "plan_ilosUtilizationByPlan";
+  }
+};
+
+const availableItems = (items: AnyObject[], entityType: string) => {
+  const updatedItemChoices: AnyObject[] = [];
+  items.forEach((item) => {
+    updatedItemChoices.push({
+      ...item,
+      label: item.name,
+      checked: false,
+      ...(entityType === "ilos" && {
+        children: [
+          {
+            id: `${parentFieldName(entityType)}_${item.id}`,
+            type: "number",
+            validation: {
+              type: "number",
+              nested: true,
+              parentFieldName: `${parentFieldName(entityType)}`,
```

Comment:
since the `parentFieldName` function now only applies to ilos, we could remove that function and go back to hardcoding the id? or put it in a constant?